File: packages/contract-npm/src/scope.ts
import type { ScopeExt } from "./models";

export interface ContractScope {
  type: string;
  asgi: Record<string, unknown>;
  scheme: string;
  http_version?: string | null;
  method?: string | null;
  path: string;
  query_string?: string;
  headers: [string, string][];
  client?: [string, number] | null;
  server?: [string, number] | null;
  ext: ScopeExt;
}

File: packages/contract-py/src/tigr_asgi_contract/scope.py
from pydantic import BaseModel
from .models import ScopeExt


class ContractScope(BaseModel):
    type: str
    asgi: dict
    scheme: str
    http_version: str | None = None
    method: str | None = None
    path: str
    query_string: bytes = b""
    headers: list[tuple[bytes, bytes]]
    client: tuple[str, int] | None = None
    server: tuple[str, int] | None = None
    ext: ScopeExt
